pkg/volume/csi/csi_plugin.go

```diff
@@ -775,11 +765,70 @@ func (p *csiPlugin) skipAttach(driver string) (bool, error) {
 	return false, nil
 }
 
+// supportsVolumeMode checks whether the CSI driver supports a volume in the given mode.
+// An error indicates that it isn't supported and explains why.
+func (p *csiPlugin) supportsVolumeMode(driver string, volumeMode storage.CSIVolumeMode) error {
+	inlineEnabled := utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume)
+	if volumeMode == storage.EphemeralVolumeMode && !inlineEnabled {
```

Comment:
I think following the switch logic below will be simpler if we do all "feature disabled" checking up here.  So something like:
```
if !inlineEnabled {
  if volumeMode != storage.PersistentVolumeMode {
    return err
  }
  return nil
}

Reply:
Done.